ext/ffi/jit_trampoline.rs

```diff
@@ -83,20 +86,35 @@ pub(crate) fn codegen(sym: &crate::Symbol) -> String {
     c += native_arg_to_c(ty);
     let _ = write!(c, " p{i}");
   }
+  if needs_unwrap {
+    let _ = write!(c, ", struct FastApiTypedArray* const p_ret");
+  }
   c += ") {\n";
-  // return func(p0, p1, ...);
-  c += "  return func(";
-  for (i, ty) in sym.parameter_types.iter().enumerate() {
-    if i > 0 {
-      c += ", ";
-    }
-    if matches!(ty, NativeType::Pointer) {
-      let _ = write!(c, "p{i}->data");
-    } else {
-      let _ = write!(c, "p{i}");
+  // func(p0, p1, ...);
+  let mut call_s = String::from("func(");
+  {
+    for (i, ty) in sym.parameter_types.iter().enumerate() {
+      if i > 0 {
+        call_s += ", ";
+      }
+      if matches!(ty, NativeType::Pointer) {
+        let _ = write!(call_s, "p{i}->data");
+      } else {
+        let _ = write!(call_s, "p{i}");
+      }
     }
+    call_s += ");\n";
   }
-  c += ");\n}\n\n";
+  if needs_unwrap {
+    // <return_type> r = func(p0, p1, ...);
+    // ((<return_type>*)p_ret->data)[0] = r;
+    let _ = write!(c, " {ffi_ret} r = {call_s}");
```

Comment:
Needs two spaces before `{ffi_ret}` I think?

Reply:
Why?

Reply:
That's how the non-wrapped path is indented so consistency.